tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,690 @@
+
+    def test_resolve_hook_with_callable_hook_returns_hook(self) -> None:
+        """_resolve_hook should return the hook when it's a configured callable."""
+        # Create a configured mock using public APIs.
+        configured_hook = MagicMock(return_value=True)
+        configured_hook.side_effect = None
+        configured_hook.reset_mock()
+
+        resolved_is_closing: list[bool] = []
+
+        def _discard_invalidated_connected_client(
+            *_args: Any,
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*_args`

(ANN401)